Rubberduck.Parsing/Rewriter/MemberAttributesRewriter.cs

```diff
@@ -32,16 +32,25 @@ public override void Rewrite()
             if(!IsDirty) { return; }
 
             var component = Module.Parent;
+            var vbeKind = component.VBE.Kind;
             if (component.Type == ComponentType.Document)
             {
                 // can't re-import a document module
                 return;
             }
 
-            var file = _exporter.Export(component);
+            var file = vbeKind == VBEKind.Embedded
```

Comment:
hmm, the enum kind of defeats the `IVBE` abstraction's purpose, no? I mean, this is pretty much `if (component.VBE is VB6.VBE) { ... }` or `if (component.VBE is VBA.VBE) { ... }`, right?

I can live with that (it's not like we're going to be adding more VBE types anytime soon), however conceptually it makes me wonder if the rewriter wouldn't be better off with two implementations, i.e. `VbaMemberAttributesRewriter` and `Vb6MemberAttributesRewriter`, and the IoC wired-up to inject the correct implementation accordingly.

Reply:
But VB6.VBE / VBA.VBE would require a hard reference to the concrete assemblies, which we definitely need to avoid. The idea of the enum was to allow divergent functionality without referencing directly, which I think we'll definitely need going forward.

Having said that, in this particular case, separate rewriter implementations are definitely cleaner, will implement.